File: src/main/java/responses/pin/Creator.java
package responses.pin;

public class Creator {
    private final String url;
    private final String firstName;
    private final String lastName;
    private final String id;

    public Creator(String url, String firstName, String lastName, String id) {
        this.url = url;
        this.firstName = firstName;
        this.lastName = lastName;
        this.id = id;
    }

    public String getUrl() {
        return url;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public String getId() {
        return id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Creator creator = (Creator) o;

        if (url != null ? !url.equals(creator.url) : creator.url != null) return false;
        if (firstName != null ? !firstName.equals(creator.firstName) : creator.firstName != null) return false;
        if (lastName != null ? !lastName.equals(creator.lastName) : creator.lastName != null) return false;
        return !(id != null ? !id.equals(creator.id) : creator.id != null);

    }

    @Override
    public int hashCode() {
        int result = url != null ? url.hashCode() : 0;
        result = 31 * result + (firstName != null ? firstName.hashCode() : 0);
        result = 31 * result + (lastName != null ? lastName.hashCode() : 0);
        result = 31 * result + (id != null ? id.hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "Creator{" +
                "url='" + url + '\'' +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", id='" + id + '\'' +
                '}';
    }
}

File: src/main/java/com/chrisdempewolf/pinterest/responses/pin/Creator.kt
package com.chrisdempewolf.pinterest.responses.pin

import com.google.gson.annotations.SerializedName

class Creator(val url: String?, @SerializedName("first_name")
val firstName: String?, @SerializedName("last_name")
              val lastName: String?, val id: String?) {

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val creator = o as Creator?

        if (if (url != null) url != creator!!.url else creator!!.url != null) return false
        if (if (firstName != null) firstName != creator.firstName else creator.firstName != null) return false
        if (if (lastName != null) lastName != creator.lastName else creator.lastName != null) return false
        return !if (id != null) id != creator.id else creator.id != null

    }

    override fun hashCode(): Int {
        var result = if (url != null) url.hashCode() else 0
        result = 31 * result + if (firstName != null) firstName.hashCode() else 0
        result = 31 * result + if (lastName != null) lastName.hashCode() else 0
        result = 31 * result + if (id != null) id.hashCode() else 0
        return result
    }

    override fun toString(): String {
        return "Creator{" +
                "url='" + url + '\'' +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", id='" + id + '\'' +
                '}'
    }
}
